apis/projectcontour/v1/httpproxy.go

```diff
@@ -1168,3 +1171,34 @@ type HTTPProxyList struct {
 	metav1.ListMeta `json:"metadata"`
 	Items           []HTTPProxy `json:"items"`
 }
+
+// SlowStart will gradually increase amount of traffic to a newly added endpoint.
+// It works only with RoundRobin and WeightedLeastRequest load balancing strategies.
+type SlowStart struct {
+	// The duration of slow start window.
+	// Duration is expressed in the Go [Duration format](https://godoc.org/time#ParseDuration).
+	// Valid time units are "ns", "us" (or "µs"), "ms", "s", "m", "h".
+	// +required
+	// +kubebuilder:validation:Pattern=`^(((\d*(\.\d*)?h)|(\d*(\.\d*)?m)|(\d*(\.\d*)?s)|(\d*(\.\d*)?ms)|(\d*(\.\d*)?us)|(\d*(\.\d*)?µs)|(\d*(\.\d*)?ns))+)$`
+	Window string `json:"window"`
+
+	// The speed of traffic increase over the slow start window.
+	// Defaults to 1.0, so that endpoint would get linearly increasing amount of traffic.
+	// When increasing the value for this parameter, the speed of traffic ramp-up increases non-linearly.
+	// The value of aggression parameter should be greater than 0.0.
+	//
+	// More info: https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/upstream/load_balancing/slow_start
+	//
+	// +optional
+	// +kubebuilder:default=`1.0`
+	// +kubebuilder:validation:Pattern=`^([0-9]+([.][0-9]+)?|[.][0-9]+)$`
+	Aggression string `json:"aggression"`
+
+	// The minimum percentage of origin weight that avoids too small new weight, which may cause endpoints in slow start mode receive no traffic in slow start window.
+	// If not specified, the default is 10%.
```

Comment:
Is this basically saying that the endpoint will get at least 10% of traffic, from the very beginning? I know this is copied mostly verbatim from envoy docs, but I think we can probably make it a little more clear, just need to read up a bit more to make sure I'm understanding exactly what it means.

Reply:
Yes exactly.  This was clearly visible when I tested with only 2 replicas and set 50% as `MinimumWeightPercent` with long 30s `window`: the pod immediately got 50% of the traffic after container restart.